src/sentry/integrations/vsts/integration.py

```diff
@@ -0,0 +1,119 @@
+    key = 'vsts'
+    name = 'VSTS'
+    metadata = metadata
+    domain = '.visualstudio.com'
+    api_version = '4.1'
+
+    setup_dialog_config = {
+        'width': 600,
+        'height': 800,
+    }
+
+    identity_oauth_scopes = frozenset([
+        'vso.code_full',
+        'vso.identity_manage',
+        'vso.work_full',
+    ])
+
+    def get_pipeline_views(self):
+        identity_pipeline_config = {
+            'oauth_scopes': self.identity_oauth_scopes,
+            'redirect_url': absolute_uri('/extensions/vsts/setup/'),
+        }
+
+        identity_pipeline_view = NestedPipelineView(
+            bind_key='identity',
+            provider_key='vsts',
+            pipeline_cls=IdentityProviderPipeline,
+            config=identity_pipeline_config,
+        )
+
+        return [identity_pipeline_view]
+
+    def get_config(self):
+        return [
+            {
+                'name': 'instance',
+                'label': 'Instance',
+                'type': 'text',
+                'placeholder': 'example.visualstudio.com',
+                'required': True,
+                'help': 'VS Team Services account ({account}.visualstudio.com) or TFS server ({server:port}).',
+            },
+            {
+                'name': 'default_project',
+                'label': 'Default Project Name',
+                'type': 'text',
+                'placeholder': 'MyProject',
+                'required': False,
+                'help': (
+                    'Enter the Visual Studio Team Services project name that you wish '
+                    'to use as a default for new work items'
+                ),
+            },
+        ]
+
+    def get_projects(self, instance, access_token):
+        session = http.build_session()
+        url = 'https://%s/DefaultCollection/_apis/projects' % instance
+        response = session.get(
+            url,
+            headers={
+                'Content-Type': 'application/json',
+                'Authorization': 'Bearer {}'.format(access_token),
+            }
+        )
+        response.raise_for_status()
+        response_json = response.json()
+        return response_json
+
+    def get_default_project(self, name, projects):
+        for project in projects:
+            if project['name'] == name:
+                return project
+
+    def build_integration(self, state):
+        data = state['identity']['data']
+        access_token = data['access_token']
+        instance = state['identity']['instance']
+        default_project_name = state['identity']['default_project']
+
+        scopes = sorted(self.identity_oauth_scopes)
+        projects = self.get_projects(instance, access_token)['value']
+        default_project = self.get_default_project(default_project_name, projects)
+        return {
+            'name': default_project['name'],
+            'external_id': default_project['id'],
+            'metadata': {
+                'access_token': access_token,
```

Comment:
The integration won't have an access token, it will have a default identity that it can use, since the access token is associated to a user, *not* the integration itself.

Reply:
Good catch. Thanks!